pkg/bytecode/vm_test.go

```diff
@@ -28,12 +25,16 @@ func TestVMGlobals(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			tt.input += "\nx = x"
 			program, err := parser.Parse(tt.input, parser.Builtins{})
 			assert.NoError(t, err, "parser error")
 			comp := NewCompiler()
 			err = comp.Compile(program)
 			assert.NoError(t, err, "compiler error")
-			vm := NewVM(comp.Bytecode())
+			bytecode := comp.Bytecode()
+			assertInstructions(t, tt.expectedInstructions, bytecode.Instructions)
```

Comment:
This looks like it was missing before and is not part of the refactor. Could you note in the commit message that some missing assertions were added too?